portfolio/src/main/webapp/Controller.js

```diff
@@ -0,0 +1,77 @@
+import CommentHandler from '/CommentHandler.js';
+import UserHandler from '/UserHandler.js';
+
+export default class Controller {
+
+  constructor(document) {
+    this.username = undefined;
+    this.loggedIn = false;
+    this.document = document;
```

Comment:
This should likely be a private instance variable, `this.document_`. I'm not 100% on this, but if you're only using the document variable internally to the Controller, which is looks like you are now, then make it private. A common pattern at Google is to have a `Controller.prototype.getDocument()` or similar method that will return private variables, if they are needed. So you could add that in the future if it turns out you do need it elsewhere.

Reply:
I didn't look in depth at `this.username` and `this.loggedIn` but follow a similar pattern as `this.document` as applicable, I'll leave that up to you.